boot/freeldr/freeldr/ntldr/arch/i386/winldr.c

```diff
@@ -64,46 +64,6 @@ typedef struct
 #define TYPE_CODE   (0x10 | DESCRIPTOR_CODE | DESCRIPTOR_EXECUTE_READ)
 #define TYPE_DATA   (0x10 | DESCRIPTOR_READ_WRITE)
 
-FORCEINLINE
-PKGDTENTRY
-KiGetGdtEntry(
-    IN PVOID pGdt,
-    IN USHORT Selector)
-{
-    return (PKGDTENTRY)((ULONG_PTR)pGdt + (Selector & ~RPL_MASK));
-}
-
-FORCEINLINE
-VOID
-KiSetGdtDescriptorBase(
-    IN OUT PKGDTENTRY Entry,
-    IN ULONG32 Base)
-{
-    Entry->BaseLow = (USHORT)(Base & 0xffff);
-    Entry->HighWord.Bytes.BaseMid = (UCHAR)((Base >> 16) & 0xff);
-    Entry->HighWord.Bytes.BaseHi  = (UCHAR)((Base >> 24) & 0xff);
-    // Entry->BaseUpper = (ULONG)(Base >> 32);
-}
-
-FORCEINLINE
-VOID
-KiSetGdtDescriptorLimit(
-    IN OUT PKGDTENTRY Entry,
-    IN ULONG Limit)
-{
-    if (Limit < 0x100000)
-    {
-        Entry->HighWord.Bits.Granularity = 0;
-    }
-    else
-    {
-        Limit >>= 12;
-        Entry->HighWord.Bits.Granularity = 1;
-    }
-    Entry->LimitLow = (USHORT)(Limit & 0xffff);
-    Entry->HighWord.Bits.LimitHi = ((Limit >> 16) & 0x0f);
-}
-
 VOID
 KiSetGdtEntryEx(
```

Comment:
Even if not really needed for the current state of this PR at least, it may be of interest to also move those next two functions into the helper header as well.

Also, second question: You moved these functions into one of NTOSKRNL's own _**private**_ headers. Not sure this is best practice here. It would be better to keep kernel's own headers for its own consumption, and using other headers that are shared with other part of code, so as to be able to use these functions elsewhere, like in freeldr as they were originally.

Reply:
It's matching where amd64 puts them as that was another suggestion made.

Reply:
OK then let's keep these there, for the time being.